### Upoznajte se s datasetom. Uočite koji su podatci ulazne, a koji izlazne veličine, ako postoje.

In [7]:
from sklearn.datasets import load_wine
data = load_wine(as_frame=True)
X = data.data
y = data.target

print(data.DESCR)

.. _wine_dataset:

Wine recognition dataset
------------------------

**Data Set Characteristics:**

:Number of Instances: 178
:Number of Attributes: 13 numeric, predictive attributes and the class
:Attribute Information:
    - Alcohol
    - Malic acid
    - Ash
    - Alcalinity of ash
    - Magnesium
    - Total phenols
    - Flavanoids
    - Nonflavanoid phenols
    - Proanthocyanins
    - Color intensity
    - Hue
    - OD280/OD315 of diluted wines
    - Proline
    - class:
        - class_0
        - class_1
        - class_2

:Summary Statistics:

============================= ==== ===== ======= =====
                                Min   Max   Mean     SD
============================= ==== ===== ======= =====
Alcohol:                      11.0  14.8    13.0   0.8
Malic Acid:                   0.74  5.80    2.34  1.12
Ash:                          1.36  3.23    2.36  0.27
Alcalinity of Ash:            10.6  30.0    19.5   3.3
Magnesium:                    70.0 162.0    99.7  14.3

In [5]:
data.frame.columns

Index(['alcohol', 'malic_acid', 'ash', 'alcalinity_of_ash', 'magnesium',
       'total_phenols', 'flavanoids', 'nonflavanoid_phenols',
       'proanthocyanins', 'color_intensity', 'hue',
       'od280/od315_of_diluted_wines', 'proline', 'target'],
      dtype='object')

In [6]:
data.frame['target']

0      0
1      0
2      0
3      0
4      0
      ..
173    2
174    2
175    2
176    2
177    2
Name: target, Length: 178, dtype: int64

Podatkovni skup sastoji se od 13 numeričkih značajki i kategoričke izlazne varijable "target". Nema nedostajućih vrijednosti. Izlazna varijabla ima vrijednosti 0, 1, 2. Ima 178 unosa.

### Na temelju znanja o navedenim algoritmima i na temelju svojstava promatranog podatkovnog skupa obrazložite koji od predloženih algoritama su prikladni za učenje i predviđanje sorte vina na temelju ostalih značajki te koji su nedostatci ostalih gore navedenih algoritama. Predložite na koji način se dataset može prilagoditi za učenje pojedinim algoritmom.

1) linearna regresija
- izlazna varijabla je kategorička stoga nam ovaj algoritam nije pogodan
- mogli bismo ga iskoristiti kada bismo npr. određivali postotak alkohola u vinu jer je on numerička varijabla

2) binarna logistička regresija
- radimo klasifikaciju, ali ne binarnu, pa nam ovaj algoritam nije koristan
- kada bismo imali samo dvije moguće vrijednosti za izlaznu varijablu mogli bismo ga koristiti

3) višeklasna klasifikacija logističkom regresijom
- ovaj algoritam bismo mogli iskoristiti za naš problem jer imamo višeklasnu klasifikaciju

4) KNN
- možemo iskoristiti ovaj algoritam jer se on može koristiti za višeklasnu klasifikaciju
- trebao bi dati dobre rezultate ako se dobro odabere broj susjeda

5) SVM
- može se koristiti za višeklasnu klasifikaciju pa ga možemo upotrijebiti
- kvaliteta njegovog predviđanja ovisit će ponajviše o odabiru hiperparametara

6) K-Means
- ovaj algoritam koristi se za nenadzirano učenje, a mi imamo izlaznu varijablu, pa nam nije idealan
- ipak koristi se za grupiranje podataka, pa iako vjerojatno nećemo dobiti baš one grupe koje odgovaraju našim klasama, bilo bi ga moguće iskoristiti 